File: backend/app/schemas/article.py
"""Article schemas for API responses."""

from datetime import datetime

from pydantic import BaseModel


class ArticleResponse(BaseModel):
    id: int
    source: str
    source_url: str | None
    title: str
    summary: str | None
    author: str | None
    published_at: datetime | None
    scraped_at: datetime
    is_processed: bool
    event_category: str | None
    tickers: list[str] = []

    model_config = {"from_attributes": True}


class PaginationMeta(BaseModel):
    page: int
    per_page: int
    total: int
    total_pages: int


class PaginatedArticles(BaseModel):
    data: list[ArticleResponse]
    meta: PaginationMeta

File: frontend/src/types/article.ts
export interface Article {
  id: number;
  source: string;
  source_url: string | null;
  title: string;
  summary: string | null;
  author: string | null;
  published_at: string | null;
  scraped_at: string;
  is_processed: boolean;
  event_category: string | null;
  duplicate_group_id: number | null;
}

export interface EventCategorySummary {
  category: string;
  count: number;
}
